From ae1fd443bada1d1b6bc65c601bff401310f3bd70 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:23 -0400
Subject: [powerpc] Remove dead code in sycall entry

Message-id: <1410545655-205645-395-git-send-email-dzickus@redhat.com>
Patchwork-id: 94291
O-Subject: [RHEL7 PATCH 394/626] powerpc: Remove dead code in sycall entry
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit fa5c11b790f9e18ec6f69680c4aa728157aef5f5
Author: Michael Neuling <mikey@neuling.org>
Date:   Fri Mar 28 16:40:39 2014 +1100

    powerpc: Remove dead code in sycall entry

    In:
      commit 742415d6b66bf09e3e73280178ef7ec85c90b7ee
      Author: Michael Neuling <mikey@neuling.org>
      powerpc: Turn syscall handler into macros

    We converted the syscall entry code onto macros, but in doing this we
    introduced some cruft that's never run and should never have been added.

    This removes that code.

    Signed-off-by: Michael Neuling <mikey@neuling.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/exceptions-64s.S b/arch/powerpc/kernel/exceptions-64s.S
index d9c650e..3afd391 100644
--- a/arch/powerpc/kernel/exceptions-64s.S
+++ b/arch/powerpc/kernel/exceptions-64s.S
@@ -54,14 +54,6 @@ END_FTR_SECTION_IFSET(CPU_FTR_REAL_LE)    \
  xori r12,r12,MSR_LE ;    \
  mtspr SPRN_SRR1,r12 ;     \
  rfid ;  /* return to userspace */  \
- b . ;      \
-2: mfspr r12,SPRN_SRR1 ;     \
- andi. r12,r12,MSR_PR ;    \
- bne 0b ;      \
- mtspr SPRN_SRR0,r3 ;     \
- mtspr SPRN_SRR1,r4 ;     \
- mtspr SPRN_SDR1,r5 ;     \
- rfid ;       \
  b . ; /* prevent speculative execution */
 
 #if defined(CONFIG_RELOCATABLE)
-- 
1.7.1